src/plugins/typebox.ts

```diff
@@ -18,11 +18,19 @@ const schemas: TSchema[] = [];
 /**
  * Register a schema in the Fastify instance.
  * This function needs to be called before the server starts, so the schema will be registered in the Fastify instance.
- * Each schema needs to have defined the `$id` property.
+ * Each schema needs to have a unique `$id` property.
  * @param schema The schema to be registered.
+ * @param options The options for the schema.
  * @returns The schema passed as argument.
  */
-export function registerSchemaAsRef<T extends TSchema>(schema: T): TRef<T> {
+export function registerSchemaAsRef<T extends TSchema>(
+  schema: T,
+  options: { $id: string; title: string },
```

Comment:
I would reverse the args, or even split like `id, title, schema` unless options are optional? But my feeling is that you usually give an id, then a title, and finally the content.